### PR TITLE
[TS-284] Re-integrate username

### PR DESCRIPTION
**Pull request by tungleu**
Brought back username, User can also login with either username or email 
image

### CODE REVIEW

**Comment by GadoliniumPotassium** [comment]
Is this implemented in a way where we can just refer to a props to make each page unique?

**Review by Ming424 — DISMISSED** [review]
lgtm, it will definitely facilitate the end user

File: src/components/Login/Login.vue [hunk 1]
```diff
@@ -4,8 +4,8 @@
       <v-form ref="formLogin" v-model="valid" data-cy="login-form">
         <v-text-field
           data-cy="email"
-          v-model="email"
-          label="Email"
+          v-model="field"
+          label="Email or username"
```
Comment (itsShaVawn) [inline src/components/Login/Login.vue:8]:
```suggestion
          label="Email or Username"
```
Capitalize this

**Review by itsShaVawn — CHANGES_REQUESTED** [review]

**Review by itsShaVawn — CHANGES_REQUESTED** [review]
You forgot to add the username to the Other Dashboard, Following, Followers and Messages pages. 
Check Figma to see where the username is on those pages.

**Comment by itsShaVawn** [comment]
I tried logging into the test user's account with the username and it does not work.
image
image
this ⬆️ does nothing when I click login.
image
also tried this ⬆️

**Comment by itsShaVawn** [comment]
I signed up with a new account and the homepage doesn't display the username
image
I also cannot log back in with the username
image

File: src/components/SignUp/SignUp.vue [hunk 2]
```diff
@@ -28,6 +28,15 @@
           </v-flex>
         </v-layout>
 
+        <v-text-field
+            data-cy="userName"
```
Comment (AlyaNaseer) [inline src/components/SignUp/SignUp.vue:32]: This doesn't follow our coding convention of "firstword-nextword"
```suggestion
            data-cy="user-name"
```

**Review by AlyaNaseer — CHANGES_REQUESTED** [review]

File: src/components/SignUp/SignUp.vue [hunk 3]
```diff
@@ -99,6 +109,10 @@
             v
           ) || "E-mail must be valid",
       },
+      rulesUsername: {
+        min: (v) => v.length >= 4 || "Min 4 characters",
+        format: (v) => /^(?!.*\.\.)(?!.*\.$)[^\W][\w.]{0,29}$/.test(v) || "No special characters"
```
[the remark this answers is not in the document]
Comment (tungleu) [reply src/components/SignUp/SignUp.vue:114]: the format accepted is a string with a minimum of 4 characters without special characters 🤔

[the remark this answers is not in the document]
Comment (AlyaNaseer) [reply src/components/SignUp/SignUp.vue:114]: Yeah I know, it's more for if our stakeholders or people look at our code in the future, but if you think it's clear enough, we can leave as-is

**Review by AlyaNaseer — APPROVED** [review]
LGTM

**Review by AlyaNaseer — DISMISSED** [review]

File: src/components/FollowerFollowing/UserBlock.vue [hunk 4]
```diff
@@ -21,8 +21,21 @@
...
         xl="3"
-        class="text-left text-body-2"
-      > {{ this.name }}
+        class="text-left mx-0 px-0 mt-3 py-0"
+      >
+        <v-list-item color="black">
+          <v-list-item-content class="py-1">
+            <v-list-item-title
+              class="text-sm-body-2 text-md-body-2 text-lg-body-2 text-xl-body-2 text-caption text-wrap"
+              >{{ this.name }}</v-list-item-title
```
Comment (itsShaVawn) [inline src/components/FollowerFollowing/UserBlock.vue:30]: need to fix indentation here

File: src/components/FollowerFollowing/UserBlock.vue [hunk 5]
```diff
@@ -21,8 +21,21 @@
...
+            >
+            <v-list-item-title
+                class="text-sm-body-2 text-md-body-2 text-lg-body-2 text-xl-body-2 text-caption text-wrap"
+            >@{{this.username}}</v-list-item-title
```
Comment (itsShaVawn) [inline src/components/FollowerFollowing/UserBlock.vue:34]: here as well

File: src/components/FollowerFollowing/UserBlock.vue [hunk 6]
```diff
@@ -21,8 +21,21 @@
...
+            >
+          </v-list-item-content>
+
```
Comment (itsShaVawn) [inline src/components/FollowerFollowing/UserBlock.vue:37]: remove extra line

File: src/components/FollowerFollowing/UserBlock.vue [hunk 7]
```diff
@@ -21,8 +21,21 @@
...
+        </v-list-item>
       </v-col>
       <v-spacer></v-spacer>
```
Comment (itsShaVawn) [inline src/components/FollowerFollowing/UserBlock.vue:40]: use self closing tag here

File: src/components/FollowerFollowing/UserBlock.vue [hunk 8]
```diff
@@ -60,13 +73,14 @@
 <script>
 import {useFollowMixin} from "../../hooks/useFollowMixin.js";
 export default {
-  
+
```
Comment (itsShaVawn) [inline src/components/FollowerFollowing/UserBlock.vue:76]: no space here

File: src/components/Login/Login.vue [hunk 9]
```diff
@@ -53,24 +53,36 @@ export default {
...
+          };
+        } else {
+          payload = {
+            username: this.field,
+            password: this.password
+          };
+        }
+        try {
+          await login(payload, this.$store, this.$router);
+        }catch (e) {
```
Comment (itsShaVawn) [inline src/components/Login/Login.vue:81]: add a space before catch

File: src/components/SignUp/SignUp.vue [hunk 10]
```diff
@@ -28,6 +28,15 @@
...
         </v-layout>
 
+        <v-text-field
+            data-cy="user-name"
+            v-model="username"
+            label="Username"
+            color="primary"
+            :rules="[rules.required, rulesUsername.min, rulesUsername.format]"
+            @keyup.enter="submit"
+        ></v-text-field>
```
Comment (itsShaVawn) [inline src/components/SignUp/SignUp.vue:38]: use a self closing tag + no extra spaces before or after the tags

File: src/components/SignUp/SignUp.vue [hunk 11]
```diff
@@ -105,22 +115,31 @@ export default {
...
           password: this.password,
           passwordConfirmation: this.passwordConfirm,
           firstname: this.firstName,
           lastname: this.lastName,
+          username: this.username
         };
-        return signUp(credentials, this.$store, this.$router);
+        try {
+          await signUp(credentials, this.$store, this.$router)
+        }catch (e){
```
Comment (itsShaVawn) [inline src/components/SignUp/SignUp.vue:140]: space before catch

File: src/hooks/useCredential.js [hunk 12]
```diff
@@ -3,44 +3,43 @@ import UserService from "../services/User.service";
...
+    });
   } else {
-    return response.message;
+    throw response.message;
   }
-  return true;
+  return false;
 };
 
+
```
Comment (itsShaVawn) [inline src/hooks/useCredential.js:15]: remove extra line

File: src/hooks/useCredential.js [hunk 13]
```diff
@@ -3,44 +3,43 @@ import UserService from "../services/User.service";
...
 export const login = async (credentials, store, router) => {
-  store
-    .dispatch("login", {
-      email: credentials.email,
-      password: credentials.password,
-    })
+
```
Comment (itsShaVawn) [inline src/hooks/useCredential.js:17]: remove extra line

File: src/hooks/useCredential.js [hunk 14]
```diff
@@ -3,44 +3,43 @@ import UserService from "../services/User.service";
...
   });
-}
+};
 
-export const brokerageSignUp = async (credentials) => {
+export const brokerageSignUp = async credentials => {
   console.log("hooks/useCredential.sj");
-  console.log("Brokerage: ", credentials.select, "\n", "Email: ", credentials.email);
-}
+  console.log(
```
Comment (itsShaVawn) [inline src/hooks/useCredential.js:36]: remove console.logs

**Review by itsShaVawn — CHANGES_REQUESTED** [review]

**Review by itsShaVawn — APPROVED** [review]

**Review by AlyaNaseer — APPROVED** [review]

**Pull request merged**